### PR TITLE
Add option to delete an item

### PR DESCRIPTION
**Pull request by betrisey**
https://user-images.githubusercontent.com/1939395/116618350-3ad4f200-a93f-11eb-8e5b-26ee936b62bf.mp4

### CODE REVIEW

File: app/src/androidTest/java/com/example/sharingang/Utils.kt [hunk 1]
```diff
@@ -51,3 +52,12 @@ fun waitAfterSaveItem() {
...
     Thread.sleep(1000)
 }
+
+fun tryOpenOptionsMenu() {
+    try {
+        Espresso.openActionBarOverflowOrOptionsMenu(InstrumentationRegistry.getInstrumentation().targetContext)
+    } catch (e: Exception) {
+        // Don't have an overflow in the action bar
+    }
+}
```
Comment (bafavre) [inline app/src/androidTest/java/com/example/sharingang/Utils.kt:62]: Nice addition!

File: app/src/main/java/com/example/sharingang/items/ItemDao.kt [hunk 2]
```diff
@@ -19,4 +19,7 @@ interface ItemDao {
 
     @Insert(onConflict = OnConflictStrategy.REPLACE)
     fun insert(items: List<Item>)
-}
\ No newline at end of file
+
+    @Query("DELETE FROM item")
```
Comment (bafavre) [inline app/src/main/java/com/example/sharingang/items/ItemDao.kt:23]: No need to specify the item id?

**Review by bafavre — APPROVED** [review]
This was truly needed, good job! :)

File: app/src/main/java/com/example/sharingang/items/ItemDao.kt (hunk 2 above)
Comment (betrisey) [reply app/src/main/java/com/example/sharingang/items/ItemDao.kt:23]: Here we want to clear the table entirely
It fixes the issue where the items are deleted on Firebase but stay in the cache.

Comment (bafavre) [reply app/src/main/java/com/example/sharingang/items/ItemDao.kt:23]: Oh yeah I see, makes sense.

**Review by Superdoude1999 — APPROVED** [review]
Nice

**Pull request merged**